mula/scheduler/server/models/task.py

```diff
@@ -29,26 +29,48 @@ class TaskStatus(str, enum.Enum):
     CANCELLED = "cancelled"
 
 
-# NOTE: model added for support of partial updates
 class Task(BaseModel):
+    id: uuid.UUID
+    scheduler_id: str
+    schedule_id: uuid.UUID | None
+    priority: int | None
+    status: TaskStatus | None
+    type: str
+    hash: str | None
+    data: dict | None
+    created_at: datetime
+    modified_at: datetime
+
+
+class TaskCreate(BaseModel):
     model_config = ConfigDict(from_attributes=True, use_enum_values=True)
 
+    # FIXME: pushing the same item with the id will update the item on the
+    # queue. Perhaps TaskCreate is not the right name for this and should
+    # be TaskPush instead, or in both cases TaskUpdate
     id: uuid.UUID | None = None
```

Comment:
We need to have the id present in the model, this is done to support the functionality of pushing the same task on the queue will update it. Meaning, pushing the same item with the id will update the item on the queue. Perhaps TaskCreate is not the right name for this and should be TaskPush instead, or in both cases TaskUpdate.

We can also opt for moving away from this functionality. Since it is a feature for queues in order to update the same item by pushing it on again this doesn't seem to be a feature that is used. Since it's inception we've moved away from an in-memory queue, and updating the task can be done from other avenues.